holoviews/plotting/bokeh/util.py

```diff
@@ -148,27 +156,166 @@ def compute_plot_size(plot):
         else:
             w_agg, h_agg = (np.max, np.sum)
         widths, heights = zip(*[compute_plot_size(child) for child in plot.children])
-        width, height = w_agg(widths), h_agg(heights)
+        return w_agg(widths), h_agg(heights)
     elif isinstance(plot, (Figure, Chart)):
-        width, height = plot.plot_width, plot.plot_height
+        if plot.plot_width:
+            width = plot.plot_width
+        else:
+            width = plot.frame_width + plot.min_border_right + plot.min_border_left
+        if plot.plot_height:
+            height = plot.plot_height
+        else:
+            height = plot.frame_height + plot.min_border_bottom + plot.min_border_top
+        return width, height
     elif isinstance(plot, (Plot, DataTable, Spacer)):
-        width, height = plot.width, plot.height
-    return width, height
+        return plot.width, plot.height
+    else:
+        return 0, 0
+
+
+def compute_layout_properties(
```

Comment:
That is one long function indeed! Glad it is now split out from the rest.

Reply:
I suspect some simplification could be done here but that doesn't have to hold up this PR...